Rocket.Chat/Managers/Socket/Response/SocketHandlers.swift

```diff
@@ -78,8 +78,19 @@ extension SocketManager {
 
             switch collection {
             case "users":
-                let user = Realm.getOrCreate(User.self, primaryKey: identifier, values: fields)
-                Realm.update(user)
+                if msg == .Added || msg == .Changed {
+                    let user = Realm.getOrCreate(User.self, primaryKey: identifier, values: fields)
+                    Realm.update(user)
+                }
+
+                if msg == .Removed {
+                    let user = Realm.getOrCreate(User.self, primaryKey: identifier, values: fields)
+
+                    Realm.execute({ (realm) in
+                        user.status = .offline
+                        realm.add(user, update: true)
+                    })
+                }
```

Comment:
@rakaramos @laggedHero First change I'll made is remove the Realm dependency from this file, creating the initial interactions with DB domain to update the objects.